ods_ci/build/install_idp_interop.sh

```diff
@@ -0,0 +1,149 @@
+  oc patch oauth cluster --type json -p '[{"op": "add", "path": "/spec/identityProviders", "value": []}]'
+fi
+
+# Patch in the HTPASSWD identityProviders
+oc patch oauth cluster --type json -p '[{"op": "add", "path": "/spec/identityProviders/-", "value": '"$OAUTH_HTPASSWD_JSON"'}]'
+
+$PROVIDER_PATH/htpasswd/htpasswd_installation.sh
+}
+
+install_ldap_identity_provider(){
+
+# Test if any oauth identityProviders exists. If not, initialize the identityProvider list
+CURRENT_IDP_LIST=$(oc get oauth cluster -o json | jq -e '.spec.identityProviders')
+if [[ -z "${CURRENT_IDP_LIST}" ]] || [[  "${CURRENT_IDP_LIST}" == "null" ]]; then
+  echo 'No oauth identityProvider exists. Initializing oauth .spec.identityProviders = []'
+  oc patch oauth cluster --type json -p '[{"op": "add", "path": "/spec/identityProviders", "value": []}]'
+fi
+
+# Patch in the LDAP identityProviders
+oc patch oauth cluster --type json -p '[{"op": "add", "path": "/spec/identityProviders/-", "value": '"$OAUTH_LDAP_JSON"'}]'
+
+$PROVIDER_PATH/ldap/ldap_installation.sh
+}
+
+add_groups_users() {
+# create groups
+oc adm groups new rhods-admins
+oc adm groups new rhods-users
+oc adm groups new rhods-noaccess
+oc adm groups new dedicated-admins
+# add users to groups
+function add_users_to_groups(){
+for i in {1..4}
+  do
+    oc adm groups add-users $1 $2$i 
+  done
+}
+add_users_to_groups rhods-admins htpasswd-admin
+add_users_to_groups rhods-users htpasswd-user
+add_users_to_groups rhods-noaccess htpasswd-noaccess
+add_users_to_groups rhods-admins ldap-admin
+add_users_to_groups dedicated-admins ldap-admin
+add_users_to_groups rhods-users ldap-user
+add_users_to_groups rhods-noaccess ldap-noaccess
+oc adm groups add-users dedicated-admins htpasswd-cluster-admin-user
+
+oc adm groups add-users rhods-admins kubeadmin
+oc adm policy add-cluster-role-to-group view rhods-admins
+oc adm policy add-cluster-role-to-group cluster-admin dedicated-admins
+
+oc describe oauth.config.openshift.io/cluster
+}
+
+function htpasswd_installation(){
+  chk_htpasswd=1
+
+  while read -r line; do
+
+    if [[ $line == *"htpasswd-cluster-admin"* ]]; then
+        echo -e "\033[0;33m Htpasswd Identity provider is installed. Skipping installation \033[0m"
+        chk_htpasswd=0
+        break
+    fi
+  done < <(oc get oauth -o yaml)
+
+  if [[ $chk_htpasswd == 1 ]]; then
+    install_htpasswd_identity_provider
+  fi
+}
+
+function ldap_installation(){
+  chk_ldap=1
+  while read -r line; do
+    if [[ $line == *"ldap-provider-qe"* ]]; then
+	    echo -e "\033[0;33m LDAP Identity provider is installed. Skipping installation \033[0m"
+	    chk_ldap=0
+        break
+    fi
+  done < <(oc get oauth -o yaml)
+  if [[ $chk_ldap == 1 ]]; then
+    install_ldap_identity_provider
+  fi
+}
+
+function createHtpasswdIDP(){
+    htpasswd -b -B -c $ARTIFACT_DIR/users.txt htpasswd-cluster-admin-user rhodsPW#123456
+    oc delete secret htpasswd-bind-password --ignore-not-found -n openshift-config
```

Comment:
IIUC the desired flow, maybe we should split logic to `cleanupHtpasswdIDP` and `createHtpasswdIDP` instead of consecutive create/delete resources call with oc??

Reply:
Hi @liswang89 why we are not reusing the existing script at
https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/build/install_idp.sh ?

Reply:
or ate least this one
https://gitlab.cee.redhat.com/ods/ods-install#install-ldap-and-htpasswd-in-openshift

Reply:
It is better to not remove it.